src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import { useContext, useEffect, useRef, useState } from 'react';
+import { ErrorMessage } from '../../enums/ErrorMessage';
+import { AuthContext } from '../Auth/AuthContext';
+
+type Props = {
+  isAllFinished: boolean,
+  setError: React.Dispatch<React.SetStateAction<ErrorMessage | null>>,
+  addTodo: (title: string) => void,
+  isAdding: boolean,
+};
+
+export const NewTodo: React.FC<Props> = ({
+  isAllFinished,
+  setError,
+  addTodo,
+  isAdding,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const newTodoField = useRef<HTMLInputElement | null>(null);
+  const user = useContext(AuthContext);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!todoTitle) {
+      setError(ErrorMessage.EMPTY);
+      return;
+    }
+
+    addTodo(todoTitle.trim());
+    setTodoTitle('');
+  };
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [user]);
```

Comment:
autofocusing after adding doesn't work. try to add `isAdding` to the dependency array

Reply:
and ask in the chat if it won't work